packages/eas-cli/src/commands/env/create.ts

```diff
@@ -0,0 +1,189 @@
+  getOwnerAccountForProjectIdAsync,
+} from '../../project/projectUtils';
+import { confirmAsync } from '../../prompts';
+import {
+  promptVariableEnvironmentAsync,
+  promptVariableNameAsync,
+  promptVariableValueAsync,
+} from '../../utils/prompts';
+
+export default class EnvironmentVariableCreate extends EasCommand {
+  static override description =
+    'create an environment variable on the current project or owner account';
+
+  static override flags = {
+    name: Flags.string({
+      description: 'Name of the variable',
+    }),
+    value: Flags.string({
+      description: 'Text value or the variable',
+    }),
+    link: Flags.boolean({
+      description: 'Link shared variable to the project',
+    }),
+    ...EASVariableSensitiveFlag,
+    ...EASVariableScopeFlag,
+    ...EASEnvironmentFlag,
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    let {
+      flags: {
+        name,
+        value,
+        scope,
+        'non-interactive': nonInteractive,
+        environment,
+        sensitive,
+        link,
+      },
+    } = await this.parse(EnvironmentVariableCreate);
+    const {
+      privateProjectConfig: { projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(EnvironmentVariableCreate, {
+      nonInteractive,
+    });
+
+    const [projectDisplayName, ownerAccount] = await Promise.all([
+      getDisplayNameForProjectIdAsync(graphqlClient, projectId),
+      getOwnerAccountForProjectIdAsync(graphqlClient, projectId),
+    ]);
+
+    if (!name) {
+      name = await promptVariableNameAsync(nonInteractive);
+    }
+
+    if (!value) {
+      value = await promptVariableValueAsync(nonInteractive);
+    }
+
+    if (scope === EnvironmentVariableScope.Project) {
+      if (!environment) {
+        environment = await promptVariableEnvironmentAsync(nonInteractive);
+      }
+      const { appVariables: existingVariables } = await EnvironmentVariablesQuery.byAppIdAsync(
+        graphqlClient,
+        projectId,
+        environment
+      );
+      const existingVariable = existingVariables.find(variable => variable.name === name);
+
+      if (existingVariable) {
+        if (!nonInteractive) {
+          await confirmAsync({
+            message: `Variable ${name} already exists on this project. Do you want to overwrite it?`,
+          });
+        }
+
+        if (existingVariable.scope === EnvironmentVariableScope.Shared) {
+          await EnvironmentVariableMutation.unlinkSharedEnvironmentVariableAsync(
+            graphqlClient,
+            existingVariable.id,
+            projectId,
+            environment
+          );
+          Log.withTick(
+            `Unlinking shared variable ${chalk.bold(name)} on project ${chalk.bold(
+              projectDisplayName
+            )}.`
+          );
```

Comment:
I believe we should ask user if they want to do it before we do so

Reply:
We do, but we don't specify that it will be unlinked.
<img width="804" alt="image" src="https://github.com/expo/eas-cli/assets/1774298/527f0a13-f1e0-4bf5-bcd4-cdd403b31a9b">
<img width="804" alt="image" src="https://github.com/expo/eas-cli/assets/1774298/a8f2d42a-5ff0-49f3-b9fe-71989f024a40">